### PR TITLE
Add Edgemark for scalability test

### PR DESCRIPTION
**Pull request by wackxu**
**What type of PR is this?**

**What this PR does / why we need it**:

Add Edgemark which is a performance testing tool which allows users to run experiments on simulated clusters

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

/assign @fisherxu @kevin-wangzefeng 

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by fisherxu — COMMENTED** [review]
Need to gofmted the `[edge/cmd/edgemark/hollow_edgecore.go]` and fix the failed DCO test :)

**Review by fisherxu — COMMENTED** [review]
/approve
/cc @zc2638 PTAL :) Edgemark for scalability test.

**Comment by zc2638** [comment]
image
I tried to build the image to run from the code, but it run failed with `cgroup`, is there something wrong with my build?

**Comment by zc2638** [comment]
Run Command:
`docker run -it harbor.daocloud.cn/zc/edgemark --token=92ae298e43ed2d0f48ec8a8ebca573e16ec85db31d49da6e557d7deda067f384.eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NDY0Mjk0NDl9.9nIw_SU335buL_qtaOUtJ2oVIr-ZZc-1sA3I8rAWnio --http-server=https://10.31.7.1:10002 --websocket-server=10.31.7.1:10000`

**Comment by wackxu** [comment]
> 

we need  privileged container to run the edgemark. you can add  `--privileged=true` and run again.

**Comment by zc2638** [comment]
> > 
> 
> we need privileged container to run the edgemark. you can add `--privileged=true` and run again.

thx.

File: edge/cmd/edgemark/hollow_edgecore.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	return edgeCoreConfig
+}
+
+func GetFakeRuntimeAndImageServices(
+	remoteRuntimeEndpoint,
+	remoteImageEndpoint string,
+	runtimeRequestTimeout metav1.Duration) (internalapi.RuntimeService, internalapi.ImageManagerService, error) {
+	endpoint, err := fakeremote.GenerateEndpoint()
+	if err != nil {
+		klog.Fatalf("Failed to generate fake endpoint %v.", err)
```
Comment (zc2638) [inline edge/cmd/edgemark/hollow_edgecore.go:140]: `error` has been thrown, we should not use `klog.fatalf` here to end the process

File: edge/cmd/edgemark/hollow_edgecore.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	// overWrite config
+	edgeCoreConfig.DataBase.DataSource = "/edgecore.db"
+	edgeCoreConfig.Modules.EdgeHub.Token = config.Token
+	edgeCoreConfig.Modules.EdgeHub.HTTPServer = config.HTTPServer
+	edgeCoreConfig.Modules.EdgeHub.WebSocket.Server = config.WebsocketServer
+
+	// use remote runtime for test, not run Dockershim
+	edgeCoreConfig.Modules.Edged.RuntimeType = "remote"
+	edgeCoreConfig.Modules.Edged.RemoteRuntimeEndpoint = "/run/containerd/containerd.sock"
+	edgeCoreConfig.Modules.Edged.RemoteImageEndpoint = "/run/containerd/containerd.sock"
```
Comment (zc2638) [inline edge/cmd/edgemark/hollow_edgecore.go:125]: `GetFakeRuntimeAndImageServices` This method uses fake runtime, so there is no need to consider the runtime endpoint here?
And I suggest that the `RuntimeType` can be changed to `fake`. Will it lead to new problems?

File: edge/cmd/edgemark/hollow_edgecore.go (hunk 1 above)
Comment (wackxu) [reply edge/cmd/edgemark/hollow_edgecore.go]: updated

File: edge/cmd/edgemark/hollow_edgecore.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+func GetFakeRuntimeAndImageServices(
+	remoteRuntimeEndpoint,
+	remoteImageEndpoint string,
+	runtimeRequestTimeout metav1.Duration) (internalapi.RuntimeService, internalapi.ImageManagerService, error) {
+	endpoint, err := fakeremote.GenerateEndpoint()
+	if err != nil {
+		klog.Errorf("Failed to generate fake endpoint %v.", err)
+		return nil, nil, err
+	}
```
Comment (zc2638) [inline edge/cmd/edgemark/hollow_edgecore.go:142]:
```suggestion
	if err != nil {
		return nil, nil, fmt.Errorf("Failed to generate fake endpoint: %v.", err)
	}
```
```golang
func Register(e *v1alpha1.Edged) {
	edgedconfig.InitConfigure(e)
	edged, err := newEdged(e.Enable)
	if err != nil {
		klog.Errorf("init new edged error, %v", err)
		os.Exit(1)
	}
	core.Register(edged)
}
```
After `edged` initialization fails, an error will be logged. Wouldn't it be better to use `fmt.Errorf()` here?

File: edge/cmd/edgemark/hollow_edgecore.go (hunk 2 above)
Comment (wackxu) [reply edge/cmd/edgemark/hollow_edgecore.go]: thanks for your review, change RuntimeType to fake and RemoteRuntimeEndpoint  to `/run/fake/fake.sock`,  becase the default value for RemoteRuntimeEndpoint  and RemoteImageEndpoint  is `unix:///var/run/dockershim.sock`, if we do not update it, 
the function `startDockerServer` will be called and edged exit due to error

https://github.com/kubeedge/kubeedge/blob/f6b6118b0f802d8a359187fab54137d1c2dc8349/edge/pkg/edged/edged.go#L653-L656

Comment (zc2638) [reply edge/cmd/edgemark/hollow_edgecore.go]: okey

File: edge/cmd/edgemark/hollow_edgecore.go (hunk 3 above)
Comment (wackxu) [reply edge/cmd/edgemark/hollow_edgecore.go]: done

**Comment by zc2638** [comment]
Okey, please update the DCO finally.

**Pull request merged**